git-archive-all.sh

```diff
@@ -246,8 +248,9 @@ fi
 if [ $VERBOSE -eq 1 ]; then
     echo -n "archiving submodules..."
 fi
+git submodule >>$TMPLIST
 while read path; do
-    TREEISH=$(git submodule | grep "^ .*${path%/} " | cut -d ' ' -f 2) # git submodule does not list trailing slashes in $path
+    TREEISH=$(cat "$TMPLIST" | grep "^ .*${path%/} " | cut -d ' ' -f 2) # git submodule does not list trailing slashes in $path
```

Comment:
This feels like a [useless use of cat](https://en.wikipedia.org/wiki/Cat_%28Unix%29#Useless_use_of_cat). Can we use `grep "^ .*${path%/} " "$TMPLIST"` instead?